R/derive_var_bcvacritxfl.R

```diff
@@ -198,7 +198,7 @@ derive_var_bcvacritxfl <- function(dataset,
   # Identify first value of x to be used for CRITx/CRITxFL
   if (is.null(critxfl_index)) {
     # Find largest index of CRITxFL already present in the dataset
-    critxfl_vars <- names(dataset)[grepl("^CRIT.*FL$", names(dataset))]
+    critxfl_vars <- names(dataset)[grepl("^CRIT.*FL$", names(dataset))] #nolint
```

Comment:
Shouldn't `grepl()` be replaced by `str_detect()`?

Reply:
Decided to leave this one as the function is superseded anyway so wanted to keep changes to it to a minimum.

Reply:
Yes, we'd previously superseded in line with admiral guidance at the time, but given the switch back to suggesting deprecation, I agree we can revisit and deprecate instead. I've made an [issue](https://github.com/pharmaverse/admiralophtha/issues/282) for it, thanks for the suggestion.